spec/app/domain/authentication/authn-jwt/input_validation/validate_claim_name_spec.rb

```diff
@@ -0,0 +1,230 @@
+      end
+    end
+
+    context "When claim name contains 1 forbidden character '*'" do
+      subject do
+        ::Authentication::AuthnJwt::InputValidation::ValidateClaimName.new().call(
+          claim_name: "*"
+        )
+      end
+
+      it "raises an error" do
+        expect { subject }.to raise_error(Errors::Authentication::AuthnJwt::FailedToValidateClaimForbiddenClaimName)
+      end
+    end
+  end
+
+  context "Valid claim name value" do
+    context "When claim name contains 1 allowed char 'F'" do
+      subject do
+        ::Authentication::AuthnJwt::InputValidation::ValidateClaimName.new().call(
+          claim_name: "F"
+        )
+      end
+
+      it 'does not raise error' do
+        expect { subject }.not_to raise_error
+      end
+    end
+
+    context "When claim name contains 1 allowed char 'f'" do
+      subject do
+        ::Authentication::AuthnJwt::InputValidation::ValidateClaimName.new().call(
+          claim_name: "f"
+        )
+      end
+
+      it 'does not raise error' do
+        expect { subject }.not_to raise_error
+      end
+    end
+
+    context "When claim name contains 1 allowed char '_'" do
+      subject do
+        ::Authentication::AuthnJwt::InputValidation::ValidateClaimName.new().call(
+          claim_name: "_"
+        )
+      end
+
+      it 'does not raise error' do
+        expect { subject }.not_to raise_error
+      end
+    end
+
+    context "When claim name contains 1 allowed char '$'" do
+      subject do
+        ::Authentication::AuthnJwt::InputValidation::ValidateClaimName.new().call(
+          claim_name: "$"
+        )
+      end
+
+      it 'does not raise error' do
+        expect { subject }.not_to raise_error
+      end
+    end
+
+    context "When claim name contains digits in the middle" do
+      subject do
+        ::Authentication::AuthnJwt::InputValidation::ValidateClaimName.new().call(
+          claim_name: "$"
+        )
+      end
+
+      it 'does not raise error' do
+        expect { subject }.not_to raise_error
+      end
+    end
+
+    context "When claim name ends with digits" do
+      subject do
+        ::Authentication::AuthnJwt::InputValidation::ValidateClaimName.new().call(
+          claim_name: "$2w"
+        )
+      end
+
+      it 'does not raise error' do
+        expect { subject }.not_to raise_error
+      end
+    end
+  end
+
+  context "Claim name exists in deny list" do
+    context "When claim name value is 'exp'" do
+      subject do
+        ::Authentication::AuthnJwt::InputValidation::ValidateClaimName.new(
+          deny_claims_list_value: ::Authentication::AuthnJwt::MANDATORY_CLAIMS_DENY_LIST
+        ).call(
+          claim_name: "exp"
+        )
+      end
+
+      it "raises an error" do
+        expect { subject }.to raise_error(Errors::Authentication::AuthnJwt::FailedToValidateClaimClaimNameInDenyList)
+      end
+    end
+
+    context "When claim name value is 'aud'" do
```

Comment:
Worth adding scenario for each forbidden claim name.. to make sure someone won't delete one in the future by mistake
Also, the same as tamir, I think worth adding a test that will not raise an error when the claim name is contain a substring of forbidden claim name. i.e. claim_name: "expref"